djautotask/api_rest.py

```diff
@@ -327,37 +327,50 @@ def get_headers(self):
 
         return headers
 
-    def _format_request_body(self, api_entity, changed_fields):
+    def _format_changed_fields(self, api_entity, changed_fields):
         body = {
             'id': api_entity.id
         }
 
         for field, value in changed_fields.items():
-
             if field in api_entity.EDITABLE_FIELDS:
                 key = api_entity.EDITABLE_FIELDS[field]
+                body = self._format_request_body(body, key, value)
 
-                if isinstance(value, datetime.datetime):
-                    body.update({
-                        key: value.astimezone(
-                            pytz.timezone('UTC')).strftime(
-                                "%Y-%m-%dT%H:%M:%SZ")
-                    })
-
-                elif isinstance(value, models.Model):
-                    body.update({
-                        key: value.id
-                    })
-
-                elif isinstance(value, decimal.Decimal):
-                    body.update(
-                        {key: str(value) if value else '0'}
-                    )
+        return body
 
-                else:
-                    body.update(
-                        {key: str(value) if value else ''}
-                    )
+    def _format_inserted_fields(self, api_entity, inserted_fields):
+        body = dict()
+
+        for field, value in inserted_fields.items():
+            if field in api_entity.AUTOTASK_FIELDS:
+                key = api_entity.AUTOTASK_FIELDS[field]
```

Comment:
Why do you have `AUTOTASK_FIELDS` and `EDITABLE_FIELDS` now? Just have the one, `EDITABLE_FIELDS` and put everything we need to edit or update in there. Then merge `_format_inserted_fields` and `_format_changed_fields` into _format_fields like we discussed on zoom. Like that example I showed you where you can merge the methods by changing the body initialization to:

`body = { 'id': api_entity.id } if api_entity else dict()`

Reply:
But that's what I spent time researching yesterday. To make sure it would be ok, and it looks like it is. If you can find a concrete reason we can't just merge them into one, then ok. But if you can't find anything please add the fields to `EDITABLE_FIELDS`. I'm fine with you making changes, but you have to have a reason for it, you can't just guess there might be a problem, you need evidence.

Reply:
Hmm.. then, it means editable fields doesn't mean "edit-allowable" fields, and it doesn't have a role of confining "edit-allowable fields". Then, my another choice could be renaming the value to `AUTOTASK_FIELDS` instead. 

Can I ask this to @kti-cameron (or @camsweeney ) again? :smiley:

Reply:
That's fine too, I just figured by keeping it EDITABLE fields it would be nicer to keep pairity with django-connectwise. I don't care what you call it as long as its merged.

Reply:
Thanks Sam, in AT case, the concept of `EDITABLE_FIELDS` is extended, so `AUTOTASK_FIELDS` could be better. The combining is applied, and ready again.